src/coreclr/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreCLR.Core.cs

```diff
@@ -0,0 +1,253 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+/*=============================================================================
+**
+**
+**
+** Purpose: Class for creating and managing a threadpool
+**
+**
+=============================================================================*/
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.ConstrainedExecution;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+    internal sealed partial class CompleteWaitThreadPoolWorkItem : IThreadPoolWorkItem
+    {
+        void IThreadPoolWorkItem.Execute() => CompleteWait();
+
+        // Entry point from unmanaged code
+        private void CompleteWait()
+        {
+            PortableThreadPool.CompleteWait(_registeredWaitHandle, _timedOut);
+        }
+    }
+
+    internal sealed class UnmanagedThreadPoolWorkItem : IThreadPoolWorkItem
```

Comment:
`UnmanagedThreadPoolWorkItem` is a dead code. You can delete it while you are on it.

Also, most of the code in this file (except the CoreCLR-specific configuration) should be moved under libraries to avoid duplication. We have missed cleaning this up when we deleted the unmanaged threadpool.